tests/test_server.py

```diff
@@ -962,3 +962,13 @@ async def test_resource_action_pagination(postgresql_client, client, clienthelpe
     response = json.loads(response.body.decode("utf-8"))
     action_ids = [uuid.UUID(resource_action["action_id"]) for resource_action in response["data"]]
     assert action_ids == third_page_action_ids
+
+
+def test_2277_typedmethod_return_optional() -> None:
```

Comment:
best to also test that the None is correctly encoded/decoded

Reply:
In the current implementation `{}` is returned instead of `{"data": null}`. I could easily change that but then `tests/test_protocol::test_return_non_warnings` fails because it asserts `"data not in response.result`. Is this assertion now obsolete, or do we observe two distinct cases, one where "data" should be absent and one where it should contain `null`?

Reply:
@wouterdb What is your opinion on this? Should I update this assertion in the failing test?

Reply:
https://jsonapi.org/format/#auto-id--top-level 

it should be null to be conform the spec, not absent. 

I think that testcase can be updated. 

(if meta is present, it data could be absent)